client/my-sites/stats/activity-log-day/index.jsx

```diff
@@ -144,47 +186,75 @@ class ActivityLogDay extends Component {
 			requestRestore,
 			rewindConfirmDialog,
 			siteId,
+			tsEndOfSiteDay,
 		} = this.props;
 
 		const hasLogs = ! isEmpty( logs );
+		const rewindHere = this.state.rewindHere;
+		const dayExpanded = this.state.dayExpanded ? true : rewindHere;
+
+		const hasConfirmDialog =
+			hasLogs &&
+			logs.some(
+				( { activityId, activityTs } ) =>
+					activityId === requestedRestoreActivityId &&
+					( tsEndOfSiteDay - DAY_IN_MILLISECONDS <= activityTs && activityTs <= tsEndOfSiteDay )
+			);
+
+		const items =
+			hasLogs &&
+			flatMap( logs, log => {
+				return [
+					log.activityId === requestedRestoreActivityId && rewindConfirmDialog,
+					<ActivityLogItem
+						applySiteOffset={ applySiteOffset }
+						disableRestore={ disableRestore }
+						hideRestore={ hideRestore }
+						key={ log.activityId }
+						log={ log }
+						requestRestore={ requestRestore }
+						siteId={ siteId }
+					/>,
+				];
+			} );
+
+		const rewindButton = hasLogs
+			? this.renderRewindButton( hasConfirmDialog ? '' : 'primary' )
+			: null;
 
 		return (
-			<div className={ classnames( 'activity-log-day', { 'is-empty': ! hasLogs } ) }>
+			<div
+				className={ classnames( 'activity-log-day', {
+					'is-empty': ! hasLogs,
+					'has-rewind-dialog': hasConfirmDialog,
+				} ) }
+			>
 				<FoldableCard
 					clickableHeader={ hasLogs }
-					expanded={ hasLogs && isToday }
-					expandedSummary={ hasLogs ? this.renderRewindButton() : null }
+					expanded={ hasLogs && ( isToday || dayExpanded ) }
+					expandedSummary={ rewindButton }
+					summary={ rewindButton }
 					header={ this.renderEventsHeading() }
 					onOpen={ this.trackOpenDay }
-					summary={ hasLogs ? this.renderRewindButton( 'primary' ) : null }
+					onClose={ this.handleCloseDay( hasConfirmDialog ) }
 				>
-					{ hasLogs &&
-						flatMap( logs, log => [
-							log.activityId === requestedRestoreActivityId && rewindConfirmDialog,
-							<ActivityLogItem
-								applySiteOffset={ applySiteOffset }
-								disableRestore={ disableRestore }
-								hideRestore={ hideRestore }
-								key={ log.activityId }
-								log={ log }
-								requestRestore={ requestRestore }
-								siteId={ siteId }
-							/>,
-						] ) }
```

Comment:
any thoughts on inlining the activities back in here? it would make it quicker to see what `render()` is rendering plus it would shrink the PR size 👍

Reply:
My bad. I've updated it now.